PerimeterXModule/PxModule.cs

```diff
@@ -189,6 +187,26 @@ private void Application_BeginRequest(object source, EventArgs e)
 				{
 					return;
 				}
+				try
+				{
+					//check if this is a telemetry command
+					if (IsTelemetryCommand(applicationContext))
+					{
+						if (IsTelemetryCommandValid(applicationContext))
+						{
+							//command is valid. send telemetry
+							PostEnforcerTelemetryActivity();
+						}
+
+						// it's not a "real" request. Stop the HTTP pipeline chain
+						application.CompleteRequest();
```

Comment:
This creates a way to disable the enforcer just by adding the right header to a request. Either put this inside the if clause above (so it only happens when the header is validated) or, preferably, don't do that at all. I don't mind if the request continues in its normal path after sending the telemetry. It's better than having a backdoor in the enforcer

Reply:
done